init.c

```diff
@@ -55,14 +55,18 @@
 #include <sys/wait.h>
 #include <sys/time.h>
 
-#define CHECK_PAGER \
-  if ((CurrentMenu == MENU_PAGER) && (idx >= 0) &&	\
-	    (MuttVars[idx].flags & R_RESORT)) \
-	{ \
-	  snprintf (err->data, err->dsize, \
-	    _("Not available in this menu.")); \
-	  return (-1); \
-	}
+static int check_pager(int idx, BUFFER *err)
+{
+  if ((CurrentMenu == MENU_PAGER) && (idx >= 0) && (MuttVars[idx].flags & R_RESORT))
+  {
+    snprintf(err->data, err->dsize, _("Not available in this menu."));
+    return (-1);
+  }
+  else
+  {
+    return 0;
+  }
+}
 
```

Comment:
Invoking check_pager as a function no longer makes the caller return. At least you'll have to turn all the calls below into

If (check_pager(...) == -1)
   return -1;

I'm not sure there's any advantage in doing so, though...